### PR TITLE
fix issue #11103

### PR DESCRIPTION
**Pull request by arthasking123**
fix issue #11103

### CODE REVIEW

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi CI tests failed, please have a look. Thanks.

**Comment by tristaZero** [comment]
@arthasking123 Thanks for the update, @strongduanmu will review it later.

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptAlterTableTokenGenerator.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
         result.addAll(getDropColumnTokens(tableName, alterTableStatementContext.getSqlStatement().getDropColumnDefinitions()));
         return result;
     }
-
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptAlterTableTokenGenerator.java:56]: @arthasking123 Keep indents consistent with the previous one. Please modify similar issues in other places.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/column/alter/ModifyColumnDefinitionSegment.java [hunk 2]
```diff
@@ -39,9 +39,11 @@
     private final int stopIndex;
     
     private final ColumnDefinitionSegment columnDefinition;
+
+    private String previousColumnName = "";
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/column/alter/ModifyColumnDefinitionSegment.java:43]: @arthasking123 Add a `ColumnDefinitionSegment newColumnDefinition` field to handle this case is better.

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/pojo/EncryptAlterTableToken.java [hunk 3]
```diff
@@ -42,6 +43,9 @@ public EncryptAlterTableToken(final int startIndex, final int stopIndex, final S
     
     @Override
     public String toString() {
+        if ("".equals(this.operationType) && "".equals(this.columnName)) {
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/pojo/EncryptAlterTableToken.java:46]: Why add this logic?

File: shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/alter_for_query_with_plain.xml [hunk 4]
```diff
@@ -41,4 +46,6 @@
         <input sql="ALTER TABLE t_account_bak DROP COLUMN password , DROP COLUMN id" />
         <output sql="ALTER TABLE t_account_bak DROP COLUMN cipher_password  , DROP COLUMN assisted_query_password  , DROP COLUMN plain_password  , DROP COLUMN id" />
     </rewrite-assertion>
+
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/alter_for_query_with_plain.xml:49]: Please remove redundant blank line.

Comment (arthasking123) [reply shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/alter_for_query_with_plain.xml:49]: @strongduanmu please check, thanks!

Comment (strongduanmu) [reply shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/alter_for_query_with_plain.xml:49]: > @strongduanmu please check, thanks!

@arthasking123 Thank you for your efforts, but you should solve the check style problem that I mentioned before.

**Comment by tristaZero** [comment]
@arthasking123 Hi thanks for your update here.
@strongduanmu Do you think it is ready to merge?

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptAlterTableTokenGenerator.java [hunk 5]
```diff
@@ -50,10 +53,43 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
...
+        } else if ("Oracle".equals(databaseName)) {
+            result.addAll(mergeDropColumnStatement(dropCollection, "(", ")"));
+        } else {
+            result.addAll(dropCollection);
+        }
         return result;
     }
     
+    private Collection<SQLToken> mergeDropColumnStatement(final Collection<SQLToken> dropCollection, final String leftJoiner, final String rightJoiner) {
+        Collection<SQLToken> filteredDropCollection = new LinkedList<>();
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptAlterTableTokenGenerator.java:69]: @arthasking123 `result` is better.

**Comment by strongduanmu** [comment]
@tristaZero After the variable naming is modified, I think this PR can be merged.

**Review by strongduanmu — APPROVED** [review]

**Comment by arthasking123** [comment]
IT was canceled

**Review by tristaZero — APPROVED** [review]
@arthasking123 Sorry for my late response. Plus, thanks for this great PR. ;-)

**Pull request merged**